lib/truffle/truffle/patching.rb

```diff
@@ -0,0 +1,38 @@
+module Truffle::Patching
+  TRUFFLE_PATCHES_DIRECTORY = File.join(Truffle::Boot.ruby_home, 'lib', 'patches')
+  TRUFFLE_PATCHES           = Dir.glob(File.join(TRUFFLE_PATCHES_DIRECTORY, '**', '*.rb')).
+      select { |path| File.file? path }.
+      map { |path| path[(TRUFFLE_PATCHES_DIRECTORY.size+1)..-4] }.
+      reduce ({}) { |h, v| h.update v => true }
```

Comment:
It does run on startup, but 3 arrays of about ten elements, that's insignificant compared to everything else. The code will also be eventually removed anyway. It's a single hash though, update does not create new instances or did you mean something else?